### PR TITLE
fix(utils): address issues in printing override warnings

### PR DESCRIPTION
**Pull request by biffgaut**
Issue 1 - Spurious overwrite warnings come up when synth’ing a stack using aws-openapigateway-lambda.

Cause – we use buildLambdaFunction() in the TemplateWriter custom resource. We fully specify the props, including the role. When overrideProps is called – this generates warnings (that have nothing to do with the client’s infrastructure or props)

Changes:
* Change ConsolidateProps() to print warnings when customer props overwrite defaults, but not when Construct props overwrite customer props (will be great as long as people used consolidateProps correctly)
* update template-writer to not send a role to buildLambaFunction, but to update the role afterwards (this is how constructs do it)

Issue 2 – deepdiff() dies. This is for 2 reasons.

Reason 1 – some properties do not have scalar values, such as maxRecordAge, which has a cdk.Duration.minutes(5) value. This causes deepdiff() to die. We have a prefilter list of values to ignore so this doesn’t happen, but if the customer does overwrite one of those values they won’t be informed.

Change– when we skip a value because it is in the prefilter list we print a warning that we cannot report on that value (is completely open with client, but will lead to noise)

Reason 2 – circular references in the props cause an infinite loop that exhausts memory requirements

Change – constructs in the props object seem to be the primary cause of this – let’s add ‘node’ to the prefilter list so deepdiff doesn’t dive into a CDK construct in props. We can print out a different warning for ‘node’ than for other prefilter list values

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 1b4c68589a4999c8c9551966743af22f0d712f8a
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2d01bb38-818e-4bd8-950d-5b3f8a23e990%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f17f7942fe1895019d26058a896a5c6b4e33243a
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8fbdfe8c-de45-48cb-bf21-385c1d5ff574%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-s3/test/test.cloudfront-s3.test.ts [hunk 1]
```diff
@@ -456,6 +456,7 @@ test("Custom resource is provisioned if encryption key is provided as bucketProp
     }
   });
   const template = Template.fromStack(stack);
+  defaults.printWarning(`******************${JSON.stringify(template)}`);
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-cloudfront-s3/test/test.cloudfront-s3.test.ts:459]: intended to leave this line?

**Review by georgebearden — DISMISSED** [review]
Looks good to me

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 526c9784af7642d6ee7b078550cc2a8d85b6c063
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=94aca043-ad9d-41e3-9c48-95a7fc1d7f3d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 9aaa229629483f6560058e0f46641072be504897
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=eb6e4041-27f0-4e04-af28-3cbedba06100%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 2c990738db86c5174e067e34c2ea45e54a77b82e
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e8b7bf1b-71e1-4afa-9677-127949255a71%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by georgebearden — DISMISSED** [review]
looks good to me

File: source/patterns/@aws-solutions-constructs/aws-constructs-factories/README.md [hunk 2]
```diff
@@ -35,19 +35,38 @@ import { Construct } from 'constructs';
 import { Stack, StackProps } from 'aws-cdk-lib';
 import { ConstructsFactories } from '@aws-solutions-constructs/aws-constructs-factories';
 
-const factories = new ConstructsFactories(this, 'integ-test');
+const factories = new ConstructsFactories(this, MyFactories');
```
Comment (hayesry) [inline source/patterns/@aws-solutions-constructs/aws-constructs-factories/README.md:38]: nit: Missing an open paren before `MyFactories`

**Review by hayesry — DISMISSED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 683a153ceeb5051395da5a1b47acbdf3adca2fb9
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=aa683df2-5d37-4784-b5d5-8e6d6366239a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by hayesry — APPROVED** [review]

**Pull request merged**